### PR TITLE
WIP (SIMP-) Update to use puppetlabs/docker 2.0.0

### PR DESCRIPTION
**Pull request by jeefberkey**
* Support puppetlabs/docker 2.0.0
* Support Puppet 5 and OEL
* Remove support for Puppet less than 4.10
* Support stdlib up to 6
* Fix default acceptance test

There is a bug regarding the group `docker::run` uses, where it uses the default from the module and not the group set in the main class. I have made an upstream ticket https://github.com/puppetlabs/puppetlabs-docker/issues/321. The acceptance test creates the `docker` group in the meantime so the tests pass.

### CODE REVIEW

**Comment by jeefberkey** [comment]
https://gitlab.com/simp/pupmod-simp-simp_docker/pipelines/29237309

**Comment by jeefberkey** [comment]
Making a PR upstream to support OEL

**Comment by jeefberkey** [comment]
https://gitlab.com/simp/pupmod-simp-simp_docker/pipelines/34319901

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -5,6 +5,8 @@
 # https://puppet.com/docs/pe/2017.3/overview/getting_support_for_pe.html#standard-releases-and-long-term-support-releases
```
Comment (op-ct) [inline .gitlab-ci.yml]:
```suggestion
 # https://puppet.com/docs/pe/2018.1/getting_support_for_pe.html#standard-releases-and-long-term-support-releases
```

**Pull request closed without merging**